internal/controllers/device_data_controller_test.go

```diff
@@ -315,6 +315,56 @@ func TestUserDevicesController_GetVehicleStatusRaw(t *testing.T) {
 	})
 }
 
+func TestParseDateRange(t *testing.T) {
+
+	for _, c := range []struct {
+		originalStart string
+		originalEnd   string
+		expectedStart string
+		expectedEnd   string
+		valid         bool
+	}{
+		{
+			originalStart: "2023-05-04",
+			originalEnd:   "2023-05-06",
+			valid:         true,
+			expectedStart: "2023-05-04",
+			expectedEnd:   "2023-05-06",
+		},
+		{
+			originalStart: "2023-05-04T09:00:00Z",
+			originalEnd:   "2023-05-06T23:00:00Z",
+			valid:         true,
+			expectedStart: "2023-05-04T09:00:00Z",
+			expectedEnd:   "2023-05-06T23:00:00Z",
+		},
+		{
+			originalStart: "",
+			originalEnd:   "",
+			valid:         true,
+			expectedStart: time.Now().Add(-1 * (time.Hour * 24 * 14)).Format(time.RFC3339),
+			expectedEnd:   time.Now().Format(time.RFC3339),
```

Comment:
This case can fail intermittently because `time.Now()` called here must land on the same second that is called in `parseDateRange`. I have run into this problem before and there is no great way around it. 
Either
1. Don't test time.Now()
2. Test that the expected Time is in a range so time.Now() + 1 minute
3. wrap time.Now() in and interface https://groups.google.com/g/golang-nuts/c/Mkonfn4j4jo

Reply:
Love too dependency inject `time.Now`